src/main/java/hudson/plugins/mercurial/MercurialInstallation.java

```diff
@@ -64,21 +64,34 @@
     private String executable;
     private boolean debug;
     private boolean useCaches;
+    private final String masterCacheRoot;
     private boolean useSharing;
     private final String config;
 
+    /** for backwards compatibility */
     @Deprecated
     public MercurialInstallation(String name, String home, String executable,
             boolean debug, boolean useCaches,
             boolean useSharing, List<? extends ToolProperty<?>> properties) {
-        this(name, home, executable, debug, useCaches, useSharing, null, properties);
+        this(name, home, executable, debug, useCaches, null, useSharing, null, properties);
     }
 
-    @DataBoundConstructor public MercurialInstallation(String name, String home, String executable, boolean debug, boolean useCaches, boolean useSharing, String config, List<? extends ToolProperty<?>> properties) {
+    /** for backwards compatibility */
```

Comment:
I mimicked how it was done when the "config" option was added. According to your suggestion, I suppose the constructor which doesn't take a "config" should also be removed, which would be an unrelated change to the feature I'm trying to add.